core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java

```diff
@@ -27,12 +28,19 @@
 import org.apache.iceberg.exceptions.NoSuchTableException;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
-import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.PropertyUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public abstract class BaseMetastoreCatalog implements Catalog {
   private static final Logger LOG = LoggerFactory.getLogger(BaseMetastoreCatalog.class);
+  private Map<String, String> catalogProps = Collections.emptyMap();
+
+  @Override
+  public void initialize(String name, Map<String, String> properties) {
+    catalogProps = properties;
+  }
```

Comment:
Rather than keeping track of properties in two places and requiring subclasses to call `super.initialize`, I think it would be better to add a new method to override, `properties`, that returns the properties map.